src/ScoreClosure.fs

```diff
@@ -53,35 +53,58 @@ let rec toScoreClosure (V: SymbolicValue) =
+                let pointFurthestToMean = if abs(w_l - mean) > abs(w_u - mean) then w_l else w_u
 
-            // The upper bound on the pdf normal is taken at either one of the ends of the interval, or at the mean (if included in the interval)
-            let normUpperApprox x y =
-                if x <= mean && mean <= y then
-                    pdfNormal (mean, sigma, mean)
+                // The function \sigma -> pdfnormal(mean, sigma, pointClosedToMean) is monotone increasing on (0, t) and monotine decreasing on (t, \infty) for some value t (which we can compute but do not need here)
+                // The minimum is thus attained at one of the two endpoints of the sigma-interval
+                min (pdfNormal (mean, sigma_l, pointFurthestToMean)) (pdfNormal (mean, sigma_u, pointFurthestToMean))
+
+            let upperBound (sigma_l, sigma_u) (w_l, w_u) =
+                if w_l <= mean && mean <= w_u then 
+                    // The mean is included in the w-interval. The maximal value is thus attained at that mean with the smallest possible sigma
+                    pdfNormal (mean, sigma_l, mean)
                 else
-                    max (pdfNormal (mean, sigma, x)) (pdfNormal (mean, sigma, y))
+                    let pointClosedToMean = if w_u < mean then w_u else w_l
+                    // We can assume that pointClosedToMean <> mean 
+                    // We know that the maximal value is attained at pointClosedToMean
 
+                    // Compute the value of sigma such that pdfnormal(mean, sigma, pointClosedToMean) is maximal
+                    let globalSigmaMax = abs (mean - pointClosedToMean)
 
-            let bounds args =
-                let (l, u) = (recRes.BoundNonLinearPart args).ToPair
-                let lower = normLowerApprox l u
-                let upper = normUpperApprox l u
+                    // Note: The function \sigma -> pdfnormal(mean, sigma, pointClosedToMean) is monotone increasing on (0, globalSigmaMax) and monote decreasing on (globalSigmaMax, \infty)
+                    if sigma_l <= globalSigmaMax && globalSigmaMax <= sigma_u then 
+                        // The value for sigma that maximises pdfnormal(mean, sigma, pointClosedToMean) is contained in the sigma-interval
+                        pdfNormal (mean, globalSigmaMax, pointClosedToMean) 
+                    else 
+                        // The max value is attained at one of the two endpoint of the sigma-interval
+                        max (pdfNormal (mean, sigma_l, pointClosedToMean) ) (pdfNormal (mean, sigma_u, pointClosedToMean) )
+
+
+            let bounds (args : list<Interval>) =
+                // Separete the bounds used by sigma and W
+                let argsForSigma = args[0..scoreClosureSigma.LinearParts.Length - 1]
+                let argsForW = args[scoreClosureSigma.LinearParts.Length..]
+
+                let sigmaBounds = (scoreClosureSigma.BoundNonLinearPart argsForSigma).ToPair

```

Comment:
I think we should ensure that `sigmaBounds` is nonnegative. I.e. setting the lower bound to 0 and erroring if the upper bound is negative.